core/corehttp/gateway_handler.go

```diff
@@ -125,6 +125,10 @@ func (i *gatewayHandler) getOrHeadHandler(w http.ResponseWriter, r *http.Request
 	i.addUserHeaders(w) // ok, _now_ write user's headers.
 	w.Header().Set("X-IPFS-Path", urlPath)
 
+	//robots-header to prevent search-engines from indexing the gateway because of the duplicate-content problematic when makeing the gateway public
+	w.Header().Set("X-Robots-Tag", "noindex")
```

Comment:
You're missing that with IPFS, the trust, auth, and ownership model changes. the gateway is _temporary_  hack to bridge to HTTP and allow stuff to be used. we're headed to native browser implementations. and this is not actually as difficult to get to as people think.

> You mentioned in #1751 to make a canonical-tag which declares the content to be owned by https://ipfs.io which I assume is as bad idea because it will destroy a part of the decentralisation ipfs should provide.

The point is not to say `ipfs.io` owns the content, but only that all IPFS content be prefixed thus when served over HTTP. Since ipfs.io will only run a gateway, anyone is free to redirect requests to their own ipfs implementations, or their own gateways. see how the Firefox and Chrome extensions work-- they reroute the requests. 

Yes, this is different than the status quo. Don't worry, policies adapt.

As for preventing indexing, i'm fine merging this _as an option_, which could default to yes. we can bear the brunt of search engines indexing at https://ipfs.io